test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -5222,42 +5263,74 @@ def save_configuration(self, outfile, mode='a'):
         """
         conf = {}
         sconf = {MGR_OBJ_SERVER: conf}
-
+        # save resourcedef file
         rd = os.path.join(self.pbs_conf['PBS_HOME'], 'server_priv',
                           'resourcedef')
-        self._save_config_file(conf, rd)
-
+        try:
+            with open(rd, 'r') as rdef_file:
+                rdef_str = rdef_file.read()
+            b64_rdef = base64.encodestring(rdef_str)
+            conf['resourcedef_file'] = b64_rdef
+        except BaseException:
+            self.logger.info("Resourcedef file not found")
+        # save pbs.conf file
+        cfg_path = os.path.join(os.sep, 'etc', 'pbs.conf')
+        with open(cfg_path, 'r') as p:
+            pbs_cfg = p.read()
+            pbs_cfg_b64 = base64.encodestring(pbs_cfg)
+        conf['pbs_conf'] = pbs_cfg_b64
+        # save hook files
+        hooks_str = self._save_hook_files()
+        conf.update(hooks_str)
+        conf['hooks'] = hooks_str
         qmgr = os.path.join(self.client_conf['PBS_EXEC'], 'bin', 'qmgr')
-
-        ret = self.du.run_cmd(self.client, [qmgr, '-c', 'p s'], sudo=True)
+        pbsnodes = os.path.join(
+            self.client_conf['PBS_EXEC'], 'bin', 'pbsnodes')
+        ret = self.du.run_cmd(
+            self.hostname, [
+                qmgr, '-c', 'print server'], sudo=True)
         if ret['rc'] != 0:
             return False
         else:
             conf['qmgr_print_server'] = ret['out']
-
-        ret = self.du.run_cmd(self.hostname, [qmgr, '-c', 'p sched'],
+        ret = self.du.run_cmd(self.hostname, [qmgr, '-c', 'print sched'],
                               sudo=True)
         if ret['rc'] != 0:
             return False
         else:
             conf['qmgr_print_sched'] = ret['out']
-
-        ret = self.du.run_cmd(self.hostname, [qmgr, '-c', 'p h'], sudo=True)
-        if ret['rc'] != 0:
-            return False
-        else:
-            conf['qmgr_print_hook'] = ret['out']
-
+        ret = self.du.run_cmd(self.hostname, [pbsnodes, '-av'])
+        nodes_val = self.utils.convert_to_dictlist(ret['out'])
+        conf['pbsnodes'] = nodes_val
         try:
-            f = open(outfile, mode)
-            cPickle.dump(sconf, f)
-            f.close()
-        except:
+            with open(outfile, mode) as f:
+                json.dump(conf, f)
+        except BaseException:
             self.logger.error('Error processing file ' + outfile)
             return False
 
         return True
```

Comment:
don't we need to save sched configuration files (sched_config, holidays, etc.? )

Reply:
Right now scope of this PR is to save only server attributes & files.
e.g, print server, print sched, hook files, pbs.conf file, pbsnodes -av.